libs/platform/src/lib/components/form/input/input.component.ts

```diff
@@ -34,12 +49,23 @@ export type InputType = 'text' | 'number' | 'email';
     selector: 'fdp-input',
     templateUrl: 'input.component.html',
     changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [{ provide: FormFieldControl, useExisting: InputComponent, multi: true }]
+    providers: [
+        { provide: FormFieldControl, useExisting: InputComponent, multi: true }
+    ]
 })
-export class InputComponent extends BaseInput {
+export class InputComponent extends BaseInput implements OnInit {
+
+    /** defines the input type of the input. */
     @Input()
     type: InputType = 'text';
 
+    state: Status | stateType = 'default';
```

Comment:
I bet already comment this one somewhere else. we should stic only with one type you dont need to import the one form core. If in core they add additional state we update ours.